### DISCUSSION TITLE
[firebase_ml_vision] Live Camera mode, with front camera Android , the smile image is painted in opposite direction in the horizontal axis.

### DISCUSSION THREAD

**Discussion by nqdung230690** [firebase/flutterfire#5567] [Feature request]
I use example in https://github.com/giandifra/Flutter-Smile-Face-Detection but in Live Camera mode, with front camera Android , the smile image is painted in opposite direction in the horizontal axis. How to fix this issuse ? Sorry my English very bad

**Comment by nqdung230690** [comment]
help me

**Comment by greenlihui** [comment]
I didn't read through your code, but I think the problem may be similar to this:
For example in the [firebase_ml_vision]:
in the [FaceDetectorPainter](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/firebase_ml_vision/example/lib/detector_painters.dart)
- when using rear(back) camera, code below works well
```dart
for (Face face in faces) {
      canvas.drawRect(
        Rect.fromLTWH(
          face.boundingBox.left * scaleX,
          face.boundingBox.top * scaleY,
          face.boundingBox.width * scaleX,
          face.boundingBox.height * scaleY,
        ),
        paint,
      );
}
```
- when using front camera, (I suppose) the image being streamed is mirrored in the screen(CameraPreivew), code below works for front camera(change left to mirrored left):
```dart
for (Face face in faces) {
      canvas.drawRect(
        Rect.fromLTWH(
          (absoluteImageSize.width - face.boundingBox.right) * scaleX,
          face.boundingBox.top * scaleY,
          face.boundingBox.width * scaleX,
          face.boundingBox.height * scaleY,
        ),
        paint,
      );
}
```
By the way, I am using `drawLTWH()` instead of `draw LTRB()` so that I only need to adjust the position of left bounding.
